api/define_alleles.py

```diff
@@ -43,52 +54,22 @@ def define_alleles(genotypedata: pd.DataFrame, locirepeats: np.ndarray, maxk: np
 			lociname_end_index += 1
 			if (lociname_end_index == len(col_names)-1):
 				locinames[lociname_index] = (prev_pos, lociname_end_index)
+	return locinames
 
-	# retrieve length of ids and  length of the list of locus names
-	nids = len(ids)
-	nloci = len(locinames)
-
-	# first section
+def _getAlleles(genotypedata: pd.DataFrame, nloci: int, locirepeats: np.ndarray, locinames: dict) -> list:
 	alleles = []
 	n = 0
 	for j in range(nloci):
 		# retrieve raw alleles (each index contains every raw alleles data with the same locinames)
 		# ex. all data with X313 prefix lociname in index 0
 		loci_name_prefix, last_index = locinames.get(j)
-		raw_alleles = []
 
-		while (n <= last_index):
-			raw_alleles += genotypedata.iloc[:, n+1].tolist()
-			n += 1
-		raw_alleles = [loci for loci in raw_alleles if str(loci) != 'nan']
+		raw_alleles, n = _getRawAlleles(genotypedata, n, last_index)
 
 		if (max(raw_alleles) - min(raw_alleles)) < locirepeats[j]:
 			# find break values(lower and upper)
-			lower_break_value = []
-			upper_break_value = []
-			counts_column = []
-
-			lower_list = []
-			upper_list = []
-
-			for allele in raw_alleles:
-				lower_list.append(allele - locirepeats[j]/2)
-				upper_list.append(allele + locirepeats[j]/2)
-
-			# search for the min from the lower_list and upper_list and add to break lists.
-			lower_break_value.append(min(lower_list))
-			upper_break_value.append(max(upper_list))
-			counts_column.append(len(lower_list))
-
-			# prepare columns of lower_bound, upper_bound, and count
-			allele_low = pd.DataFrame(lower_break_value)
-			allele_high = pd.DataFrame(upper_break_value)
-			allele_count = pd.DataFrame(counts)
-
-			# put allele columns together to make dataframe
-			allele_df = pd.concat([allele_low, allele_high, allele_count], axis=1)
-			allele_df.columns = ['lower_break_value', 'upper_break_value', 'counts']
-			alleles.append(allele_df)
+			lower_break_value, upper_break_value, counts_column = _findBreakValues(raw_alleles, locirepeats, j)
+			alleles = _prepareDataframes(alleles, lower_break_value, upper_break_value, counts)
 		else:
 			# making breaks (not sure if we need this)
```

Comment:
This else statement is still really huge; could you break it out into its own function, since it seems to be doing something different from the if clause above?